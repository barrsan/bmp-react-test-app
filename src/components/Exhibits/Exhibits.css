.exhibits {
  padding: 20px;
}