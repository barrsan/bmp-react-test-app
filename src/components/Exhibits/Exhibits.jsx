import React from 'react';
import PropTypes from 'prop-types';
import Table from '../Table/Table';
import SearchForm from '../SearchForm/SearchForm';
import FilterSelector from '../FilterSelector/FilterSelector';
import Pagination from '../Pagination/Pagination';
import './Exhibits.css';


const propTypes = {
  items: PropTypes.array,
  filterData: PropTypes.array,
  currentPage: PropTypes.number,
  setExhibitsFilter: PropTypes.func.isRequired,
  setExhibitsSearchString: PropTypes.func.isRequired,
};

const defaultProps = {
  items: [],
  filterData: [],
  currentPage: 0,
};


const Exhibits = ({
  items,
  filterData,
  currentPage,
  pagesNumber,
  setExhibitsFilter,
  setExhibitsSearchString,
}) => {
  const columns = [
    {
      header: 'Название',
      accessor: 'name',
      width: '200px',
    },
    {
      header: 'Место создания',
      accessor: 'place',
      width: '200px',
      filterComponent: (
        <FilterSelector
          type="place"
          options={filterData}
          handler={setExhibitsFilter}
        />
      ),
    },
    {
      header: 'Организация',
      accessor: 'organization',
      width: '200px',
    },
    {
      header: 'Описание',
      accessor: 'description',
      width: '',
    },
  ];

  return (
    <div className="exhibits">
      <h1>Экспонаты</h1>
      <SearchForm
        setSearchString={setExhibitsSearchString}
      />
      <Table
        columns={columns}
        data={items}
      />
      <Pagination
        currentPage={currentPage}
        pagesNumber={pagesNumber}
      />
    </div>
  );
};


Exhibits.propTypes = { ...propTypes };
Exhibits.defaultProps = { ...defaultProps };


export default Exhibits;
