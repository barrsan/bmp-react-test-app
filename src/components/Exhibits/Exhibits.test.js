import React from 'react';
import { shallow } from 'enzyme';
import Exhibits from './Exhibits';


describe('Exhibits component', () => {
  const props = {
    setExhibitsFilter: () => {},
    setExhibitsSearchString: () => {},
  };

  const wrapper = shallow(<Exhibits {...props} />);

  test('renders correctly', () => expect(wrapper).toMatchSnapshot());

  test('renders without crashing', () => {
    expect(wrapper).toHaveLength(1);
    expect(wrapper.find('SearchForm')).toHaveLength(1);
    expect(wrapper.find('Table')).toHaveLength(1);
    expect(wrapper.find('Pagination')).toHaveLength(1);
  });
});
