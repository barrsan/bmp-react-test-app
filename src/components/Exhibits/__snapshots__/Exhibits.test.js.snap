// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exhibits component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Exhibits
    currentPage={0}
    filterData={Array []}
    items={Array []}
    setExhibitsFilter={[Function]}
    setExhibitsSearchString={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Экспонаты
</h1>,
        <SearchForm
          setSearchString={[Function]}
/>,
        <Table
          columns={
                    Array [
                              Object {
                                "accessor": "name",
                                "header": "Название",
                                "width": "200px",
                              },
                              Object {
                                "accessor": "place",
                                "filterComponent": <FilterSelector
                                  handler={[Function]}
                                  options={Array []}
                                  type="place"
                    />,
                                "header": "Место создания",
                                "width": "200px",
                              },
                              Object {
                                "accessor": "organization",
                                "header": "Организация",
                                "width": "200px",
                              },
                              Object {
                                "accessor": "description",
                                "header": "Описание",
                                "width": "",
                              },
                            ]
          }
          data={Array []}
/>,
        <Pagination
          currentPage={0}
          pagesNumber={0}
/>,
      ],
      "className": "exhibits",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Экспонаты",
        },
        "ref": null,
        "rendered": "Экспонаты",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "setSearchString": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "columns": Array [
            Object {
              "accessor": "name",
              "header": "Название",
              "width": "200px",
            },
            Object {
              "accessor": "place",
              "filterComponent": <FilterSelector
                handler={[Function]}
                options={Array []}
                type="place"
/>,
              "header": "Место создания",
              "width": "200px",
            },
            Object {
              "accessor": "organization",
              "header": "Организация",
              "width": "200px",
            },
            Object {
              "accessor": "description",
              "header": "Описание",
              "width": "",
            },
          ],
          "data": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "currentPage": 0,
          "pagesNumber": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Экспонаты
</h1>,
          <SearchForm
            setSearchString={[Function]}
/>,
          <Table
            columns={
                        Array [
                                    Object {
                                      "accessor": "name",
                                      "header": "Название",
                                      "width": "200px",
                                    },
                                    Object {
                                      "accessor": "place",
                                      "filterComponent": <FilterSelector
                                        handler={[Function]}
                                        options={Array []}
                                        type="place"
                        />,
                                      "header": "Место создания",
                                      "width": "200px",
                                    },
                                    Object {
                                      "accessor": "organization",
                                      "header": "Организация",
                                      "width": "200px",
                                    },
                                    Object {
                                      "accessor": "description",
                                      "header": "Описание",
                                      "width": "",
                                    },
                                  ]
            }
            data={Array []}
/>,
          <Pagination
            currentPage={0}
            pagesNumber={0}
/>,
        ],
        "className": "exhibits",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Экспонаты",
          },
          "ref": null,
          "rendered": "Экспонаты",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "setSearchString": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "columns": Array [
              Object {
                "accessor": "name",
                "header": "Название",
                "width": "200px",
              },
              Object {
                "accessor": "place",
                "filterComponent": <FilterSelector
                  handler={[Function]}
                  options={Array []}
                  type="place"
/>,
                "header": "Место создания",
                "width": "200px",
              },
              Object {
                "accessor": "organization",
                "header": "Организация",
                "width": "200px",
              },
              Object {
                "accessor": "description",
                "header": "Описание",
                "width": "",
              },
            ],
            "data": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "currentPage": 0,
            "pagesNumber": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
